nullaway/src/main/java/com/uber/nullaway/dataflow/CoreNullnessStoreInitializer.java

```diff
@@ -92,13 +94,25 @@ private static NullnessStore lambdaInitialStore(
     Symbol.MethodSymbol fiMethodSymbol = NullabilityUtil.getFunctionalInterfaceMethod(code, types);
     com.sun.tools.javac.util.List<Symbol.VarSymbol> fiMethodParameters =
         fiMethodSymbol.getParameters();
+    // This obtains the Types of the functional interface method with preserved annotations in case
+    // of generic types
+    List<Type> overridenMethodParamTypeList =
+        types.memberType(ASTHelpers.getType(code), fiMethodSymbol).getParameterTypes();
```

Comment:
I was expecting that somewhere we would have to check the config to see if we were in JSpecify mode.  I think that coded this way, this will change NullAway's behavior outside of JSpecify mode, which could turn into a compatibility issue for existing code.  Could we change this PR so that the changes only apply in JSpecify mode?  Also maybe we should write a test to check that the NullAway behavior when JSpecify mode is off is unchanged.

Reply:
Done! Let me know if this works